spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MultiRNNCell.scala

```diff
@@ -0,0 +1,165 @@
+import scala.reflect.ClassTag
+
+/**
+ * Enable user stack multiple simple cells.
+ */
+class MultiRNNCell[T : ClassTag](val cells: Array[Cell[T]])(implicit ev: TensorNumeric[T])
+  extends Cell[T](
+    hiddensShape = cells.last.hiddensShape,
+    regularizers = cells.flatMap(_.regularizers)) {
+  // inputDim and hidDim must be the same with Recurrent
+  private val inputDim = Recurrent.inputDim
+  private val hidDim = Recurrent.hidDim
+
+  override var preTopology: TensorModule[T] = null
+
+  override var cell: AbstractModule[Activity, Activity, T] = buildModel()
+
+  def buildModel(): Sequential[T] = {
+    val seq = Sequential()
+    cells.foreach{ cell =>
+      if (cell.preTopology != null) {
+        cell.includePreTopology = true
+      }
+      seq.add(cell)
+    }
+    seq
+  }
+
+  override def updateOutput(input: Table): Table = {
+    val result = T()
+    result(inputDim) = input(inputDim)
+    val states = input(hidDim).asInstanceOf[Table]
+    val outputStates = T()
+
+    var i = 0
+    while (i < cells.length) {
+      result(hidDim) = states(i)
+      cells(i).forward(result).toTable
+      result(inputDim) = cells(i).output.toTable(inputDim)
+      outputStates(i) = cells(i).output.toTable(hidDim)
+      i += 1
+    }
+
+    result(hidDim) = outputStates
+    this.output = result
+    output
+  }
+
+  override def updateGradInput(input: Table, gradOutput: Table): Table = {
+    var i = cells.length
+    var error = T()
+    error(inputDim) = gradOutput(inputDim)
+    val states = input(hidDim).asInstanceOf[Table]
+    val gradStates = gradOutput(hidDim).asInstanceOf[Table]
+    val outputGradStates = T()
+
+    val nextInput = T()
+    while (i >= 1) {
+      val input0: Tensor[T] = if (i > 1) {
+        cells(i - 2).output.toTable(inputDim)
+      } else input(inputDim)
+      nextInput(inputDim) = input0
+
+      nextInput(hidDim) = states(i - 1)
+      error(hidDim) = gradStates(i - 1)
+      error = cells(i - 1).updateGradInput(nextInput, error)
+      outputGradStates(i - 1) = error(hidDim)
+      i -= 1
+    }
+
+    this.gradInput = error
+    gradInput(hidDim) = outputGradStates
+    gradInput
+  }
+
+  override def accGradParameters(input: Table, gradOutput: Table): Unit = {
+    var i = cells.length
+    val error = T()
+    error(inputDim) = gradOutput(inputDim)
+    val states = input(hidDim).asInstanceOf[Table]
+    val gradStates = gradOutput(hidDim).asInstanceOf[Table]
+
+    val nextInput = T()
+    while (i >= 1) {
+      val input0: Tensor[T] = if (i > 1) {
+        cells(i - 2).output.toTable(inputDim)
+      } else input(inputDim)
+      nextInput(inputDim) = input0
+
+      nextInput(hidDim) = states(i - 1)
+      error(hidDim) = gradStates(i - 1)
+      cells(i - 1).accGradParameters(nextInput, error)
+      error(inputDim) = cells(i - 1).gradInput.toTable(inputDim)
+      i -= 1
+    }
+  }
+
+  override def backward(input: Table, gradOutput: Table): Table = {
+    var i = cells.length
```

Comment:
better to set i to cells.length-1, so as to align with the updateOutput implemetation